tests/hardhat/Comptroller/XVSSpeeds.ts

```diff
@@ -0,0 +1,91 @@
+    vToken1 = await smock.fake<VToken>("VToken");
+    vToken2 = await smock.fake<VToken>("VToken");
+
+    await accessControl.isAllowedToCall.returns(true);
+    await vToken1.isVToken.returns(true);
+    await vToken2.isVToken.returns(true);
+
+    await comptroller._setAccessControl(accessControl.address);
+    await comptroller._supportMarket(vToken1.address);
+    await comptroller._supportMarket(vToken2.address);
+  });
+
+  describe("_initializeMarket", () => {
+    it("Supply and borrow state after initializing the market in the pool", async () => {
+      const supplyRate = await (await comptroller.venusSupplyState(vToken1.address)).toString().split(",")[0];
+      const borrowRate = await (await comptroller.venusBorrowState(vToken1.address)).toString().split(",")[0];
+
+      expect(supplyRate).equal(convertToUnit(1, 36));
+      expect(borrowRate).equal(convertToUnit(1, 36));
+    });
+  });
+
+  describe("_setVenusSpeeds", async () => {
+    it("Revert on invalid supplySpeeds input", async () => {
+      await expect(
+        comptroller._setVenusSpeeds(
+          [vToken1.address, vToken2.address],
+          [convertToUnit(1, 15)],
+          [convertToUnit(1, 15), convertToUnit(1, 15)],
+        ),
+      ).to.be.revertedWith("Comptroller::_setVenusSpeeds invalid supplySpeeds");
+    });
+
+    it("Revert on invalid borrowSpeeds input", async () => {
+      await expect(
+        comptroller._setVenusSpeeds(
+          [vToken1.address, vToken2.address],
+          [convertToUnit(1, 15), convertToUnit(1, 15)],
+          [convertToUnit(1, 15)],
+        ),
+      ).to.be.revertedWith("Comptroller::_setVenusSpeeds invalid borrowSpeeds");
+    });
+
+    it("Revert for unlisted market", async () => {
+      const [unListedMarket] = await ethers.getSigners();
+      await expect(
+        comptroller._setVenusSpeeds([unListedMarket.address], [convertToUnit(1, 16)], [convertToUnit(1, 15)]),
+      ).to.be.revertedWith("Comptroller:: setVenusSpeedInternal market not listed");
+    });
+
+    it("Revert on invalid borrowSpeeds input", async () => {
+      await comptroller._setVenusSpeeds(
+        [vToken1.address, vToken2.address],
+        [convertToUnit(1, 16), convertToUnit(1, 18)],
+        [convertToUnit(1, 20), convertToUnit(1, 22)],
+      );
+
+      const token1SupplySpeed = await comptroller.venusSupplySpeeds(vToken1.address);
+      const token1BorrowSpeed = await comptroller.venusBorrowSpeeds(vToken1.address);
+
+      expect(token1SupplySpeed).equal(convertToUnit(1, 16));
+      expect(token1BorrowSpeed).equal(convertToUnit(1, 20));
+
+      const token2SupplySpeed = await comptroller.venusSupplySpeeds(vToken2.address);
+      const token2BorrowSpeed = await comptroller.venusBorrowSpeeds(vToken2.address);
+
+      expect(token2SupplySpeed).equal(convertToUnit(1, 18));
+      expect(token2BorrowSpeed).equal(convertToUnit(1, 22));
+    });
+  });
```

Comment:
Should we add a test case for checking accruals when setting speed to 0 and then to a positive value?

Reply:
Added a test for the same: [commit](https://github.com/VenusProtocol/venus-protocol/pull/140/commits/e7f7411400975993903a9fd82cea9f847b333c02)